portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,10 +37,30 @@
   
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String text = getParameter(request, "text-input", "");
-    String[] comments = text.split("\\s*,\\s*");
-    response.setContentType("text/html;");
-    response.getWriter().println(Arrays.toString(comments));
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("comment", getParameter(request, "text-input", ""));
+    taskEntity.setProperty("timestamp", System.currentTimeMillis());
+    datastore.put(taskEntity);
+    
+    Query query = new Query("Task").addSort("comment", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Task> tasks = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        String comment = (String) entity.getProperty("comment");
+        long timestamp = (long) entity.getProperty("timestamp");
+
+        Task task = new Task(id, comment, timestamp);
+        tasks.add(task);
```

Comment:
This can be wrapped to one line once you finish debugging. 
tasks.add(new Task(entity.getKey().getId(), (String) entity.getProperty("comment"), (long) entity.getProperty("timestamp")));

Reply:
Just in case you want to experiment Java 8 streaming, this will be interesting to try. https://stackoverflow.com/questions/35650974/create-list-of-object-from-another-using-java-8-streams

Reply:
Done. Ill be sure to check that out, I've never messed with streaming before, could be interesting!